### PR TITLE
Use MAC from payload if not in advertised data.

### PR DESCRIPTION
**Pull request by ttu**
Fix for BLE advertised mfg_data.

### CODE REVIEW

File: ruuvitag_sensor/ruuvi.py [hunk 1]
```diff
@@ -156,17 +156,22 @@ def _get_ruuvitag_datas(macs=[], search_duratio_sec=None, run_flag=RunFlag(), bt
...
-            # If data is not valid RuuviTag data add MAC to blacklist
+            # If data is not valid RuuviTag data add MAC to blacklist if MAC is available
             if data is not None:
-                state = get_decoder(data_format).decode_data(data)
-                if state is not None:
-                    yield (ble_data[0], state)
+                decoded = get_decoder(data_format).decode_data(data)
+                if decoded is not None:
+                    # Check whitelist using MAC from decoded data if advertised MAC is not available
+                    if not ble_data[0] and decoded['mac'] and macs and not decoded['mac'] in macs:
```
Comment (sergioisidoro) [inline ruuvitag_sensor/ruuvi.py:170]: Note that `mac` in ruuvi pay load comes in a format without `:` like `112c6a1e593d`

`':'.join(mac_address[i:i+2] for i in range(0,12,2))` should make it standard like the others.

Comment (ttu) [reply ruuvitag_sensor/ruuvi.py]: Thanks for pointing it out. Made fix :+1:

**Comment by ttu** [comment]
@sergioisidoro Added fix for `mfg_data` based on your findings.

**Comment by sergioisidoro** [comment]
Awesome. I'll test this and I'll report back.

**Comment by sergioisidoro** [comment]
Hi, 

This is strange. I can only get results if I change the log level of bleson to debug...
Also I see multiple advertisements but I only see one result. 

```
Get latest data for sensors. Stop with Ctrl+C.
2020-04-11 22:21:16,229   INFO -                 ruuvi.py: 99 -     get_data_for_sensors(): Get latest data for sensors. Stop with Ctrl+C.
Stops automatically in 5s
2020-04-11 22:21:16,229   INFO -                 ruuvi.py:100 -     get_data_for_sensors(): Stops automatically in 5s
MACs: []
2020-04-11 22:21:16,229   INFO -                 ruuvi.py:101 -     get_data_for_sensors(): MACs: []
Start receiving broadcasts (device 0)
2020-04-11 22:21:16,276   INFO -                bleson.py: 56 -                    start(): Start receiving broadcasts (device 0)
2020-04-11 22:21:16,871  DEBUG -              __init__.py: 26 -             get_provider(): Provider is <bleson.providers.macos.macos_provider.MacOSProvider object at 0x10e78b650>
2020-04-11 22:21:16,873  DEBUG -         macos_adapter.py: 99 -   wait_for_event_timeout(): wait_for_event_timeout starting
2020-04-11 22:21:16,877  DEBUG -         macos_adapter.py:101 -   wait_for_event_timeout(): event set: True
2020-04-11 22:21:16,877  DEBUG -         macos_adapter.py: 95 -                      off(): TODO: adatper off
2020-04-11 22:21:16,878  DEBUG -         macos_adapter.py: 92 -                       on(): TODO: adatper on
2020-04-11 22:21:16,878  DEBUG -         macos_adapter.py:171 - centralManagerDidUpdateState_(): State: 5

<REDACTED>

2020-04-11 22:21:17,225  DEBUG -                 types.py:387 -                 __init__(): Advertisement(flags=0x06, name='None', txpower=None, uuid16s=[], uuid128s=[], rssi=None, mfg_data=None)
2020-04-11 22:21:17,225  DEBUG -         macos_adapter.py:210 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): kCBAdvDataManufacturerData=b'\xe0\x00\x01e\xca\x96(\xf7'
2020-04-11 22:21:19,336  DEBUG -         macos_adapter.py:176 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): Found: name=None rssi=-78 data={
    kCBAdvDataChannel = 37;
    kCBAdvDataIsConnectable = 0;
    kCBAdvDataManufacturerData = <99040536 340276c7 a000b4fc 4c00e0b3 b6972ef4 e2f7d664 de42>;
} 

<REDACTED>

2020-04-11 22:21:23,240  DEBUG -                 types.py:387 -                 __init__(): Advertisement(flags=0x06, name='None', txpower=None, uuid16s=[], uuid128s=[], rssi=None, mfg_data=None)
2020-04-11 22:21:23,240  DEBUG -         macos_adapter.py:192 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): kCBAdvDataServiceUUIDs:
2020-04-11 22:21:23,240  DEBUG -                 types.py:158 -                 __init__(): UUID128 type: <class 'bytes'> value=b'\x98\xed\x00\x01\xa5A\x11\xe4\xb6\xa0\x00\x02\xa5\xd5\xc5\x1b'
2020-04-11 22:21:23,241  DEBUG -                 types.py:188 -                 __init__(): UUID128('98ed0001-a541-11e4-b6a0-0002a5d5c51b')
2020-04-11 22:21:23,241  DEBUG -         macos_adapter.py:206 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): Service UUID: <objective-c class CBUUID at 0x7fffa6dff3c0> 98ED0001-A541-11E4-B6A0-0002A5D5C51B
2020-04-11 22:21:33,468  DEBUG -         macos_adapter.py:176 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): Found: name=None rssi=-77 data={
    kCBAdvDataChannel = 37;
    kCBAdvDataIsConnectable = 0;
    kCBAdvDataManufacturerData = <99040536 3a0262c7 a100acfc 4c00f0b3 b6972eff e2f7d664 de42>;
} 
2020-04-11 22:21:33,468  DEBUG -                 types.py:387 -                 __init__(): Advertisement(flags=0x06, name='None', txpower=None, uuid16s=[], uuid128s=[], rssi=None, mfg_data=None)
2020-04-11 22:21:33,469  DEBUG -         macos_adapter.py:210 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): kCBAdvDataManufacturerData=b'\x99\x04\x056:\x02b\xc7\xa1\x00\xac\xfcL\x00\xf0\xb3\xb6\x97.\xff\xe2\xf7\xd6d\xdeB'
2020-04-11 22:21:34,957  DEBUG -         macos_adapter.py:176 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): Found: name=None rssi=-74 data={
    kCBAdvDataChannel = 37;
    kCBAdvDataIsConnectable = 0;
    kCBAdvDataManufacturerData = <990405ff 244ec5c7 64ffd0ff d40434a0 361c5350 ef9a92b4 6d43>;
} 
2020-04-11 22:21:34,957  DEBUG -                 types.py:387 -                 __init__(): Advertisement(flags=0x06, name='None', txpower=None, uuid16s=[], uuid128s=[], rssi=None, mfg_data=None)
2020-04-11 22:21:34,958  DEBUG -         macos_adapter.py:210 - centralManager_didDiscoverPeripheral_advertisementData_RSSI_(): kCBAdvDataManufacturerData=b'\x99\x04\x05\xff$N\xc5\xc7d\xff\xd0\xff\xd4\x044\xa06\x1cSP\xef\x9a\x92\xb4mC'
2020-04-11 22:21:34,959  DEBUG -         macos_adapter.py:110 -            stop_scanning(): 
{'E2:F7:D6:64:DE:42': {'data_format': 5, 'humidity': 1.57, 'temperature': 69.38, 'pressure': 1011.04, 'acceleration': 990.5957803261631, 'acceleration_x': 180, 'acceleration_y': -948, 'acceleration_z': 224, 'tx_power': 4, 'battery': 3037, 'movement_counter': 151, 'measurement_sequence_number': 12020, 'mac': 'e2f7d664de42'}}
```

I wonder if it's a good idea to put things to the queue directly, instead of processing the data to a dict, and then putting it to the queue.


Ps: Redacted some of my other BLE devices. 
And yes, that's the sauna sensor, and we're about to have sauna :)

**Comment by ttu** [comment]
Added some debug logging to bleson.py, so if you can try again and send new logs.

If it still won't work, then can try to move the data handling done on _run_get_data_background to  Observer’s on_advertising_data. 

Hope you get RuuviTags to work nicely with Home Assistant. I should setup HA again and then my RuuviTags would get some proper use 👍

**Comment by ttu** [comment]
Released this as won't break any functionality that currently works. Create a new Issue for macOS problems or add to previous discussions if/when problems are found.

**Pull request merged**